pkg/providers/az-client.go

```diff
@@ -1,15 +1,32 @@
 package providers
 
 import (
+	"fmt"
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/authorization/armauthorization"
 	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/subscription/armsubscription"
+	"github.com/Azure/draft/pkg/cred"
 )
 
 type AzClient struct {
-	AzTenantClient azTenantClient
+	AzTenantClient   azTenantClient
+	RoleAssignClient *armauthorization.RoleAssignmentsClient
 }
 
 //go:generate mockgen -source=./az-client.go -destination=./mock/az-client.go .
 type azTenantClient interface {
 	NewListPager(options *armsubscription.TenantsClientListOptions) *runtime.Pager[armsubscription.TenantsClientListResponse]
 }
+
+func createRoleAssignmentClient(subscriptionId string) (*armauthorization.RoleAssignmentsClient, error) {
+	cred, err := cred.GetCred()
+	if err != nil {
+		return nil, fmt.Errorf("getting credentials: %w", err)
+	}
+
+	client, err := armauthorization.NewRoleAssignmentsClient(subscriptionId, cred, nil)
+	if err != nil {
+		return nil, fmt.Errorf("creating role assignment client: %w", err)
```

Comment:
💡 since its error code what do you think about this:

* ` fmt.Errorf("failed to create role assignment client: %w", err)`